build-trust/ockam
# Crate Release Oct 25, 2021

- ci: add ignore for bad commit message
- docs(rust): fix broken rustdoc links in ockam_node
- docs(rust): fix broken rustdoc links in ockam_entity
- ci: commit lint only on pull request and dispatch
- feat(elixir): remote forwarder worker for elixir
- build(elixir): recompile vault libs when running tests
- fix(rust): fix clippy errors in the ockam crate
- fix(rust): fix clippy errors in the ockam_entity crate
- docs(rust): fix broken rustdoc links in the ockam crate
- chore(rust): remove `None` errors from error enums
- feat: add new crate with the "node_test" attribute
- test: use it on tcp transport "send_receive" test
- chore: minor changes
- feat: add more hygiene rules
- refactor: split test and non-test macros
- test: add more test cases for new hygiene rules
- chore: use node_test in tcp test
- chore: tidy up crate files
- ci: no tab lint for stderr logs
- feat(rust): add std, no_std and alloc features to getting started examples
- chore(rust): update cargo lock files
- chore(rust): fix warnings in `ockam_node` under no_std+alloc
- ci: update token for docker builder and base
- ci: enable crate release
- chore(rust): update rust docker version
- ci: install rust nightly to docker builder
- ci: remove cargo audit from docker builder
- ci: update rust version to 1.56.0
- ci: remove all repository dispatches
- ci: bump ockam builder version
- feat(rust): move as many things as possible into a workspace
- fix(rust): ensure example lib/bin names are unique
- chore: run cargo fmt on the examples
- fix(rust): fix various rust examples
- fix(rust): fix rust target paths in gha cache
- fix(rust): move deny.toml to workspace root
- chore(rust): implement the feature structure proposed in #1883
- chore(rust): remove ockam_examples/example_projects from the workspace for now
- fix(rust): remove redundant lint checks
- fix(rust): simplify feature usage to accepted structure
- chore(rust): fixup remaining tests for workspace structure
- fix(rust): use full example names for rust kafka examples
- feat(rust): balena block addon
- fix(rust): fix portal_sender
- chore(rust): update kafka runner
- build(elixir): move default config to mix.exs file
- build(elixir): update ranch and ex_doc dependencies
- feat(rust): add examples dirs to workspace
- ci: script tcp inlet/outlet example
- build: update erlang, elixir, and jdk versions in builder
- ci: bump builder version
- ci(elixir): fix formatting errors caused by elixir version update
- test(rust): revert latest example_runner changes
- build(elixir): fix git reference for brod library in lock files
- refactor(rust): replace instances of `&Vec<T>` with `&[T]`
- build(elixir): fix git reference to brod in ockam_kafka
- build(elixir): cleanup dockerfile build instructions to not copy ffi libs
- build(elixir): roll back to brod version 3.15
- docs: fix stream guide markdown
- ci: add inlet examples to markdown verification
- feat(rust): make async-trait used through ockam_core
- feat(rust): add generic async_try_clone implementation for structs with clone
- feat(rust): make handle async only
- feat(rust): make vault traits async
- feat(rust): make ockam_vault async
- feat(rust): make ockam_vault_sync_core async
- test(rust): make vault tests async
- feat(rust): make key exchange async
- feat(rust): secure channel async updates
- feat(rust): make entity async
- feat(rust): remove block_future from tcp
- feat(rust): get_started async update
- feat(rust): ffi async updates
- feat(rust): fix zeroize usage
- refactor(rust): remove protocol parser and asyncify stream workers
- feat(rust): make delayed_event async
- feat(rust): impl async_try_clone for tcp_transport
- feat(rust): expose async_try_clone from ockam crate
- feat(rust): remove secure_channels trait
- docs: make docs and examples use async functions
- docs(rust): fix up secure channel streams example
- feat(rust): update tracing subscriber
- ci: remove one ignored rustsec issue after fixing
- ci: remove unused utility
- ci: license example_runner
- ci: unignore advisory
- ci: license examples and utilities
- chore(rust): fix clippy warnings for ockam_entity
- chore(rust): remove redundant workspace member glob and update comment
- test(rust): use random port for send_receive test
- ci: remove call to cargo bump from release script
- feat(rust): crate release oct 25 2021



## Current Behaviour



## Proposed Changes




## Checks



- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://www.ockam.io/learn/how-to-guides/contributing/CONTRIBUTING#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://www.ockam.io/learn/how-to-guides/high-performance-team/conduct/).
- [x] I have accepted the Ockam [Contributor Licence Agreement](https://www.ockam.io/learn/how-to-guides/contributing/cla/) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/ockam-network/contributors/blob/master/CONTRIBUTORS.csv) file in a separate pull request to the [ockam-network/contributors](https://github.com/ockam-network/contributors) repository.
